pkg/operator/reports.go

```diff
@@ -91,8 +91,26 @@ func (op *Reporting) syncReport(logger log.FieldLogger, key string) error {
 		}
 		return err
 	}
+
 	sr := report.DeepCopy()
 
+	// check if the report is expired
+	if report.DeletionTimestamp == nil {
+		if reportExpired := isReportExpired(logger, report, time.Now()); reportExpired {
```

Comment:
I think we may need to check if the report has reached expiration in `runReport` too, like we check if a report has already finished:

```go
func (op *Reporting) runReport(logger log.FieldLogger, report *metering.Report) error {
	// check if report has reached its expiration date. If true, stop processing
        // processing the resource, and exit earliy.
	if reportExpired := isReportExpired(logger, report, time.Now()); reportExpired {
		return nil
	}
        ...
```

Reply:
Done in 4a78406b